runner.py

```diff
@@ -274,6 +274,31 @@ def process_data(with_indels, dust, evalue, fastq_files, log, max_basecall_itera
     return reference_file
 
 
+def validate_input(output_dir, input_dir, reference_file):
+    if os.path.exists(output_dir):
+        list_dir = os.listdir(output_dir)
+        if len(list_dir) > 0:
+            if not ((len(list_dir) == 1) and (list_dir[0] == 'pbs_logs')):
+                raise Exception("output_dir must be path to a new or empty directory!")
+    if not os.path.isfile(reference_file):
+        raise Exception("Reference_file must exist!")
+    else:
+        file_name = os.fsdecode(reference_file)
+        if not file_name.endswith(".fasta"):
+            raise Exception("Reference_file must be fasta file!")
+    if not os.path.isdir(input_dir):
+        raise Exception("Input_dir must exist!")
+    input_dir_loop = os.fsencode(input_dir)
+    if len(os.listdir(input_dir_loop)) > 0:
+        for file in os.listdir(input_dir_loop):
+            file_name = os.fsdecode(file)
+            if file_name.endswith(".fastq"):
+                break
+            raise Exception("There are no fastq files in input_dir!")
```

Comment:
again you dont check the actual files but only their suffix so the error message here is not correct.
Also, is it necessary..? won't it fail in a meaningful way without this?

Reply:
Not with a meaningful exception- the parallel running is for each file in the folder- so if it is empty the error will be after and not with the correct information. After debugging,  In line 326 in runner- it parses the files, so if there are not any from the right format the function for computing is not executed – the len for fastq files – how many are there is zero, without exception being raised.

Reply:
please put in an accurate error message and use 'not' instead of 'break'